src/aruco_pose/src/aruco_pose.cpp

```diff
@@ -0,0 +1,218 @@
+    this->get_parameter("dist_coeffs.k2").as_double(),
+    this->get_parameter("dist_coeffs.p1").as_double(),
+    this->get_parameter("dist_coeffs.p2").as_double(),
+    this->get_parameter("dist_coeffs.k3").as_double(),
+    this->get_parameter("dist_coeffs.k4").as_double(),
+    this->get_parameter("dist_coeffs.k5").as_double(),
+    this->get_parameter("dist_coeffs.k6").as_double());
+
+  physical_marker_size_ = this->get_parameter("physical_marker_size").as_double();
+
+  double alpha = this->get_parameter("cam_rotation.alpha").as_double() / 180 * M_PI;
+  double beta = this->get_parameter("cam_rotation.beta").as_double() / 180 * M_PI;
+  double gamma = this->get_parameter("cam_rotation.gamma").as_double() / 180 * M_PI;
+
+  // Add the camera to tf server
+  camera_quaternion_.setRPY(alpha, beta, gamma);
+
+  // Define the transform
+  geometry_msgs::msg::TransformStamped transformStamped;
+  transformStamped.header.stamp = this->now();
+  transformStamped.header.frame_id = "world";
+  transformStamped.child_frame_id = this->get_parameter("camera_tf_frame").as_string();
+  transformStamped.transform.translation.x = this->get_parameter("cam_translation.x").as_double();
+  transformStamped.transform.translation.y = this->get_parameter("cam_translation.y").as_double();
+  transformStamped.transform.translation.z = this->get_parameter("cam_translation.z").as_double();
+  transformStamped.transform.rotation.x = this->camera_quaternion_.x();
+  transformStamped.transform.rotation.y = this->camera_quaternion_.y();
+  transformStamped.transform.rotation.z = this->camera_quaternion_.z();
+  transformStamped.transform.rotation.w = this->camera_quaternion_.w();
+
+  static_broadcaster_.sendTransform(transformStamped);
+
+  camera_subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+    "/rgb/image_raw", 5,
+    std::bind(&ArucoPose::image_raw_callback, this, std::placeholders::_1));
+
+  // Assign parameters for ArUco tag detection
+  parameters_ = cv::aruco::DetectorParameters::create();
+  parameters_->cornerRefinementMethod = cv::aruco::CORNER_REFINE_APRILTAG;
+
+  auto it = dictionary_map_.find(this->get_parameter("fiducial_marker_family").as_string());
+  if (it != dictionary_map_.end()) {
+    dictionary_ = cv::aruco::getPredefinedDictionary(it->second);
+  } else {
+    RCLCPP_ERROR(
+      LOGGER, "Invalid dictionary name: %s", this->get_parameter(
+        "fiducial_marker_family").as_string().c_str());
+    throw std::runtime_error("Invalid dictionary name");
+  }
+
+  // Initial estimates
+  median_filtered_rpyxyz = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+
+  // Configure the median filter. 6 refers to the number of channels in the multi-channel filter
+  // Note: it is necessary to have/declare a parameter named 'number_of_observations'
+  // in the parameter server.
+  median_filter_->configure(
+    6, "", "number_of_observations",
+    this->get_node_logging_interface(), this->get_node_parameters_interface());
```

Comment:
Why is this getting ros objects, how is it attending to them?

Reply:
After re-checking, filter is a ROS library. Definition of the configure function can be found in this [link](
https://github.com/ros/filters/blob/41cc2b6464752f70c95a1716e5bac00b52081c12/include/filters/filter_base.hpp#L82-L107)